sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala

```diff
@@ -645,6 +666,19 @@ abstract class CastBase extends UnaryExpression with TimeZoneAwareExpression wit
       b => x.exactNumeric.asInstanceOf[Numeric[Any]].toInt(b)
     case x: NumericType =>
       b => x.numeric.asInstanceOf[Numeric[Any]].toInt(b)
+    case DayTimeIntervalType(s, e) if s == e =>
+      buildCast[Long](_, i => dayTimeIntervalToInt(i, e))
+    case YearMonthIntervalType(s, e) if s == e =>
+      buildCast[Int](_, i => yearMonthIntervalToInt(i, e))
+  }
+
+  private[this] def dayTimeIntervalToInt(v: Long, endFiled: Byte): Int = {
```

Comment:
Could you embed it as this function is used only once.

Reply:
make sense.